### PR TITLE
ECM-662

### PR DESCRIPTION
**Pull request by hsjhita**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ECM-662

### Change description ###

Changed date format of date so it can be sorted correctly. Also added in code to sort the list each time the cause list is generated

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/ListingHelper.java [hunk 1]
```diff
@@ -100,6 +101,7 @@
             MEMBER_DAYS_REPORT, RESPONDENTS_REPORT, SESSION_DAYS_REPORT, ECC_REPORT);
     private static final List<String> SCOTLAND_HEARING_LIST = List.of("Reading Day", "Deliberation Day",
             "Members meeting", "In Chambers");
+    private static final DateTimeFormatter dateTimePattern = DateTimeFormatter.ofPattern("dd MMMM yyyy");
```
Comment (gauravchawla73463) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/ListingHelper.java:104]: Should Month format be MMM as there are months which have less than 4 characters?

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/ListingServiceTest.java [hunk 2]
```diff
@@ -43,7 +43,6 @@
 import uk.gov.hmcts.ecm.common.model.reports.respondentsreport.RespondentsReportCaseData;
```
Comment (gauravchawla73463) [inline src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/ListingServiceTest.java:43]: May be a unit test for the case when CauseListDate(s) are empty.

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/ListingHelper.java (hunk 1 above)
Comment (hsjhita) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/ListingHelper.java:104]: It shouldn't affect May. MMMM will still show May. The original was "d MMMM yyyy". I have just changed it to "dd" to allow it to be in order. It was already MMMM

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/ListingServiceTest.java (hunk 2 above)
Comment (jthmcts) [reply src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/ListingServiceTest.java:43]: Write a unit test for dates 01 January and 01 December. Which will come first?

Comment (hsjhita) [reply src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/ListingServiceTest.java:43]: @gauravchawla73463 - 2 tests created, 1 for listed date and other for causeListDate, both will cause exceptions
@jthmcts - added extra code for the comparator and a unit test to test 01 December and Jan

**Review by gauravchawla73463 — APPROVED** [review]

**Review by jthmcts — APPROVED** [review]

**Pull request merged**